apps/crn-server/test/controllers/research-outputs.test.ts

```diff
@@ -153,13 +153,58 @@ describe('ResearchOutputs controller', () => {
 
       const researchOutputCreateDataObject =
         getResearchOutputCreateDataObject();
-      expect(researchOutputDataProviderMock.create).toBeCalledWith({
-        ...researchOutputCreateDataObject,
-        addedDate: mockDate.toISOString(),
-      });
+      expect(researchOutputDataProviderMock.create).toBeCalledWith(
+        {
+          ...researchOutputCreateDataObject,
+          addedDate: mockDate.toISOString(),
+        },
+        true,
+      );
       spy.mockRestore();
     });
 
+    test.each`
+      publishedValue | situation               | expected
+      ${true}        | ${'publish is true'}    | ${true}
+      ${false}       | ${'publish is false'}   | ${false}
+      ${undefined}   | ${'publish is not set'} | ${true}
+    `(
+      'Should call data provider with published equals $expected when $situation',
+      async ({ publishedValue, expected }) => {
+        const mockDate = new Date('2010-01-01');
+        const spy = jest
+          .spyOn(global, 'Date')
+          .mockImplementation(() => mockDate);
+
+        const researchOutputCreateData = getResearchOutputCreateData();
+        const researchOutputId = 'research-output-id-1';
+        researchOutputDataProviderMock.create.mockResolvedValueOnce(
+          researchOutputId,
+        );
+
+        const result =
```

Comment:
im not a huge fan of such logic inside tests, i think would prefer a `describe` about publishing within which we would have one test to prove that when its set to false - false is passed on to the controller, and when its missing - it defaults to true

Reply:
fixed here [review: change published to create options](https://github.com/yldio/asap-hub/pull/2667/commits/b072177f672bd5f5e7ce1a2eeea1d168638ed9ce)